lizmap/server/lizmap_accesscontrol.py

```diff
@@ -0,0 +1,284 @@
+
+    def __init__(self, server_iface: 'QgsServerInterface') -> None:
+        super().__init__(server_iface)
+
+        self.iface = server_iface
+
+    def layerFilterExpression(self, layer: 'QgsVectorLayer') -> str:
+        """ Return an additional expression filter """
+        return super().layerFilterExpression(layer)
+
+    def layerFilterSubsetString(self, layer: 'QgsVectorLayer') -> str:
+        """ Return an additional subset string (typically SQL) filter """
+        return super().layerFilterSubsetString(layer)
+
+    def layerPermissions(self, layer: 'QgsMapLayer') -> QgsAccessControlFilter.LayerPermissions:
+        """ Return the layer rights """
+        # Get default layer rights
+        rights = super().layerPermissions(layer)
+
+        # Get Lizmap user groups provided by the request
+        groups = self.getLizmapGroups()
+
+        # If groups is empty, no Lizmap user groups provided by the request
+        # The default layre rights is applied
+        if len(groups) == 0:
+            return rights
+
+        # Get Lizmap config
+        cfg = self.getLizmapConfig()
+        if not cfg:
+            # Lizmap config is empty
+            QgsMessageLog.logMessage("Lizmap config is empty", "lizmap", Qgis.Warning)
+            # Default layer rights applied
+            return rights
+
+        # Check Lizmap config layers
+        if 'layers' not in cfg or not cfg['layers']:
+            # Lizmap config has no options
+            QgsMessageLog.logMessage("Lizmap config has no layers", "lizmap", Qgis.Warning)
+            # Default layer rights applied
+            return rights
+
+        # Get layers config
+        cfg_layers = cfg['layers']
+        # Get layer name
+        layer_name = layer.name()
+
+        # Check lizmap edition config
+        layer_id = layer.id()
+        if 'editionLayers' in cfg and cfg['editionLayers']:
+            if layer_id in cfg['editionLayers'] and cfg['editionLayers'][layer_id]:
+                edit_layer = cfg['editionLayers'][layer_id]
+
+                # Check if edition is possible
+                # By default not
+                canEdit = False
+                if 'acl' in edit_layer and edit_layer['acl']:
+                    # acl is defined and not an empty string
+                    # authorization defined for edition
+                    group_edit = edit_layer['acl'].split(',')
+                    group_edit = [g.strip() for g in group_edit]
+
+                    # check if a group is in authorization groups list
+                    if len(group_edit) != 0:
+                        for g in groups:
+                            if g in group_edit:
+                                canEdit = True
+                    else:
+                        canEdit = True
+                else:
+                    # acl is not defined or an empty string
+                    # no authorization defined for edition
+                    canEdit = True
```

Comment:
Is that correct? And L80 too?

Reply:
Yes it's correct, if no `acl` or if `acl` is an empty string no restriction defined.